subsys/net/ip/net_context.c

```diff
@@ -746,6 +746,52 @@ int net_context_put(struct net_context *context)
 	return ret;
 }
 
+int net_context_put_all(struct net_if *iface)
+{
+	int i;
+	int ret = 0;
+
+	k_sem_take(&contexts_lock, K_FOREVER);
+
+	for (i = 0; i < NET_MAX_CONTEXT; i++) {
+		if (!net_context_is_used(&contexts[i])) {
+			continue;
+		}
+
+		if (net_context_get_iface(&contexts[i]) != iface) {
+			continue;
+		}
+
+		/* Only TCP contexts are handled here */
+		if (net_context_get_type(&contexts[i]) != SOCK_STREAM) {
+			continue;
+		}
+
+		k_mutex_lock(&contexts[i].lock, K_FOREVER);
+
+		if (IS_ENABLED(CONFIG_NET_OFFLOAD) &&
+		    net_if_is_ip_offloaded(iface)) {
+			contexts[i].flags &= ~NET_CONTEXT_IN_USE;
+			ret = net_offload_put(iface, &contexts[i]);
+			k_mutex_unlock(&contexts[i].lock);
+			continue;
+		}
+
+		contexts[i].connect_cb = NULL;
+		contexts[i].recv_cb = NULL;
+		contexts[i].send_cb = NULL;
```

Comment:
We shouldn't clear the callbacks I think, TCP still needs `connect_cb`/`recv_cb` (depending on the internal state it's in) to notify the socket layer about connection being closed? If we clear the callback now, the application won't get an error from the socket.